### DISCUSSION TITLE
Support for multi-object (aka database) constraints

### DISCUSSION THREAD

**Discussion by autarch** [geldata/gel#2972] [Ideas]
Apologies if this is already possible, but I can't find anything in the docs that suggest that this.

It would be great if there were support for constraints spanning multiple objects of a given type. A good example is an album and its tracks. Each track has a track number. Ideally, the database would enforce that for each album, its tracks starts with `1` and continue sequentially.

If I were using Postgres, I'd use a deferred trigger that checks the state of the table, though the exact details get a bit complex.

Another way to think of this is as constraints on the entire state of the database.

It'd be great if EdgeDB supported this without having to use triggers like this.

**Comment by CodesInChaos** [comment]
Related issue https://github.com/edgedb/edgedb/issues/3089
